src/wallet/spend.cpp

```diff
@@ -889,7 +889,7 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
             coin_selection_params.tx_noinputs_size += ::GetSerializeSize(txout, PROTOCOL_VERSION);
         }
 
-        if (IsDust(txout, wallet.chain().relayDustFee())) {
+        if (!recipient.scriptPubKey.IsTrue() && IsDust(txout, wallet.chain().relayDustFee())) {
```

Comment:
probably shouldn't allow making a tx you will not propagate due to mempool flags